spec/grape/api/instance_spec.rb

```diff
@@ -51,4 +51,48 @@ def app
       expect(an_instance.top_level_setting.parent).to be_nil
     end
   end
+
+  context 'with multiple moutes' do
+    let(:root_api) do
+      first = Class.new(Grape::API::Instance) do
+        namespace(:some_namespace) do
+          route :any, '*path' do
+            error!('Not found! (1)', 404)
+          end
+        end
+      end
+      second = Class.new(Grape::API::Instance) do
+        namespace(:another_namespace) do
+          route :any, '*path' do
+            error!('Not found! (2)', 404)
+          end
+        end
+      end
+      Class.new(Grape::API) do
+        mount first
+        mount first
+        mount second
+      end
+    end
+
+    it 'does not raise a FrozenError on first instance' do
```

Comment:
I understand that the bug is a frozen error raised, but let's rewrite this as a positive test of what you expect the response to be. Otherwise in the future when the behavior changes but no exceptions are raised we won't catch this issue.

Reply:
Correct me if I'm getting you wrong, but there is a positive and a negative test?
```
it 'responds the correct body at the first instance' 
it 'does not raise a FrozenError on first instance'
```

Could you explain this in more detail please? :)

Reply:
The first covers the second in terms of functionality.

The second standalone is a misleading test. It would make sense if the call in some cases was supposed to raise FrozenError and we explicitly wanted to ensure that in certain conditions that's not the case.